### PR TITLE
Finished making a screen for the Rank Employees Use Case.

### PR DESCRIPTION
**Pull request by kaizen-w**

### CODE REVIEW

**Review by tlan4 — DISMISSED** [review]
good use of interfaces

**Review by lungerjo — APPROVED** [review]

**Review by lungerjo — APPROVED** [review]

File: src/main/java/DAO/DepartmentDAO.java [hunk 1]
```diff
@@ -133,7 +133,7 @@ public List<Department> getAllDepartments(){
                 }
 
                 // create a department
-                UUID id = UUID.fromString(departmentResult.getString("ID"));
+                UUID id = UUID.fromString(departmentResult.getString("ID").trim());
```
Comment (tlan4) [inline src/main/java/DAO/DepartmentDAO.java:136]: Why trim? Wouldn't this cause issues with the existing entries in the database (projects, employees, etc.)?

**Review by tlan4 — APPROVED** [review]
Looks good, maybe add more documentation?

File: src/main/java/DAO/DepartmentDAO.java (hunk 1 above)
Comment (kaizen-w) [reply src/main/java/DAO/DepartmentDAO.java:136]: Without trim there were issues logging in for me.

Comment (kaizen-w) [reply src/main/java/DAO/DepartmentDAO.java:136]: It's a fail-safe.

**Review by zeyuzhang516 — APPROVED** [review]

**Review by zeyuzhang516 — APPROVED** [review]

**Review by GitHaoBao — DISMISSED** [review]

**Review by lungerjo — APPROVED** [review]

**Review by tlan4 — APPROVED** [review]

**Review by tlan4 — APPROVED** [review]

**Review by staceylry — APPROVED** [review]

**Review by lungerjo — APPROVED** [review]

**Review by tlan4 — APPROVED** [review]

**Review by zeyuzhang516 — APPROVED** [review]

**Pull request merged**